src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -517,11 +564,52 @@ private void linkToStudy(String targetStudy, String sampleName, int numOfRowsToB
 
         log("Link to study: Choose target");
         selectOptionByText(Locator.id("targetStudy"), "/" + targetStudy + " (" + targetStudy + " Study)");
+        if (categoryName != null)
+            setFormElement(Locator.name("autoLinkCategory"), categoryName);
         clickButton("Next");
 
         new DataRegionTable("query", getDriver()).clickHeaderButtonAndWait("Link to Study");
     }
 
+    private void createDatasetCategory(String projectName, String name)
+    {
+        goToProjectHome(projectName);
+        goToManageViews();
+        Locator.linkWithText("Manage Categories").findElement(getDriver()).click();
+        _extHelper.waitForExtDialog("Manage Categories");
+        Window categoryWindow = new Window.WindowFinder(getDriver()).withTitle("Manage Categories").waitFor();
+        categoryWindow.clickButton("New Category", 0);
+        WebElement newCategoryField = Locator.input("label").withAttributeContaining("id", "textfield").notHidden().waitForElement(getDriver(), WAIT_FOR_JAVASCRIPT);
+        setFormElementJS(newCategoryField, name);
+        fireEvent(newCategoryField, SeleniumEvent.blur);
+        waitForElement(Ext4Helper.Locators.window("Manage Categories").append("//div").withText(name));
+        clickButton("Done", 0);
+        _extHelper.waitForExtDialogToDisappear("Manage Categories");
+    }
+
+    private String getCategory(String projectName, String datasetName)
+    {
+        goToProjectHome(projectName);
+        goToSchemaBrowser();
+        ExecuteQueryPage executeQueryPage = ExecuteQueryPage.beginAt(this, "study", "DataSets");
+        DataRegionTable table = executeQueryPage.getDataRegion();
+        table.setFilter("Label", "Equals", datasetName);
+        return table.getDataAsText(0, "categoryid");
+    }
+
+    private void deleteDatasets(String projectName)
+    {
+        goToProjectHome(projectName);
+        _studyHelper.goToManageDatasets().clickDeleteMultipleDatasets();
+        if (Locator.xpath("//table/tbody/tr").findElements(getDriver()).size() > 1)
+        {
+            checkCheckbox(Locator.xpath("//table/tbody/tr[1]/td[1]/input"));
+            clickButton("Delete Selected", 0);
+            acceptAlert().contains("Are you sure you want to delete the selected datasets? This action cannot be undone.");
+        }
+        waitFor(() -> 1 == Locator.xpath("//table/tbody/tr").findElements(getDriver()).size(), WAIT_FOR_JAVASCRIPT);
+    }
```

Comment:
Could you use TestDataGenerator for this?
Example:
`TestDataGenerator.deleteDomain(projectName, "study", "Sample type 2");`